### DISCUSSION TITLE
How to improve startup times in AWS Lambda

### DISCUSSION THREAD

**Discussion by vladaman** [quarkusio/quarkus#29553] [Q&A]
We have fairly simple REST API deployed on AWS Lambda behind AWS API Gateway HTTP. The startup times are quite high. Especially after we added Hibernate the startup time went from 2.5s to 7.5s or more.

```
2022-11-29 11:57:33,797 INFO  [io.quarkus] (main) MyApi 1.0.0-SNAPSHOT on JVM (powered by Quarkus 2.11.3.Final) started in 7.354s. 
2022-11-29 11:57:33,798 INFO  [io.quarkus] (main) Installed features: [agroal, amazon-lambda, cdi, hibernate-orm, hibernate-orm-panache, jdbc-mysql, narayana-jta, qute, rest-client, rest-client-jackson, resteasy, resteasy-jackson, security, smallrye-context-propagation, vertx]
```

application.properties:
```
%prod.quarkus.hibernate-orm.database.generation=create
%prod.quarkus.hibernate-orm.log.sql=false
```

The function.zip package size is ~70MB and deployed from AWS S3

Is there any way to speed this up? Can any of the dependencies be removed to shrink the deployment size?

![image](https://user-images.githubusercontent.com/907206/204526510-40939cb6-8069-4afb-9be2-9f01537dd2fb.png)

[a comment by an automated account is not shown]

**Comment by gsmet** [comment]
Is there any way you could get us an execution profile? A flamegraph similar to what you get with asyncprofiler could be helpful.

A few general comments:
- `%prod.quarkus.hibernate-orm.database.generation=create` will drop the database schema and recreate it every time you start the application. It is usually a slow operation. Is it really what you want?
- maybe you could try RESTEasy Reactive and see how it goes. But my guess is that, while it could improve the situation a bit, it won't fix the elephant in the room.

**Comment by vladaman** [comment]
Thanks @gsmet

1. I changed the `generation=none` with virtually no difference. Hibernate still scans the DB and outputs recommendations into console log.
2. Can you Please guide me how to generate execution profile locally or even on lambda?

**Reply by gsmet** [reply]
Hmmm, `none` shouldn't do anything so there's something incorrect going on.

We have a very complete guide here: https://github.com/quarkusio/quarkus/blob/main/TROUBLESHOOTING.md . You should at least be able to get a local profiling result.
